### ISSUE TITLE
Supporting for detection of `a[href^="#"]` referring to non existent id

### ISSUE THREAD

**Issue by kagankan** [markuplint/markuplint#649]
`no-refer-to-non-existent-id` rule warns only `for` or `aria-` attributes.

I hope the rule warns the code below:

```html
<a href="#main">To main</a>
<!-- No `main` ID -->
```

Correct examples:

```html
<a href="#main">To main</a>
<main id="main"></main>
```

```html
<a href="/other-page/#main">To other page</a>
```

The codes below will be always correct according to the spec.
https://html.spec.whatwg.org/multipage/browsing-the-web.html#scrolling-to-a-fragment

>If fragment is the empty string, then return the special value top of the document.

```html
<a href="#">To top</a>
```

>If decodedFragment is an ASCII case-insensitive match for the string top, then return the top of the document.

```html
<a href="#top">To top</a>
```

---

(edit)

After posting, I remembered `name` attribute was "obsolete but conforming"...
https://html.spec.whatwg.org/multipage/obsolete.html#obsolete-but-conforming-features

```html
<a href="#main">To main</a>
<a name="main"></a>
```

So this code should also be treated as correct code.
(Of course, `deprecated-attr` rule warns this, but the href's reference is correct.)
Maybe this feature will beyond the scope of no-refer-to-non-existent-_id_ ...? 🤔

**Comment by YusukeHirao** [comment]
@kagankan Thank you!

I consider it.

**Comment by YusukeHirao** [comment]
The `no-refer-to-non-existent-id` rule should be improved.
